plugins/modules/gitlab_merge_request.py

```diff
@@ -0,0 +1,414 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023, Ondrej Zvara (ozvara1@gmail.com)
+# Based on code:
+# Copyright (c) 2021, Lennert Mertens (lennert@nubera.be)
+# Copyright (c) 2021, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2015, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2019, Guillaume Martinez (lunik@tiwabbit.fr)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: gitlab_merge_request
+short_description: Create, update, or delete GitLab merge requests
+version_added: 7.1.0
+description:
+  - Creates a merge request if it does not exist.
+  - When a merge request does exist, it will be updated if the provided parameters are different.
+  - When a merge request does exist and O(state=absent), the merge request will be deleted.
```

Comment:
```suggestion
  - When a single merge request does exist and O(state=absent), the merge request will be deleted.
```

Reply:
I think it would avoid confusion if you specify clearly that the task will fail if multiple MR exist.